### ISSUE TITLE
v1.0 config equivalent in v2.0

### ISSUE THREAD

**Issue by logicmd** [NTMC-Community/MatchZoo#398]
Hi,

I found v2.0 no longer support training config in v1.0.

Is [categorical_crossentropy](https://github.com/faneshion/MatchZoo/blob/master/examples/toy_example/config/cdssm_classify.config#L78) equivalent to losses.rank_crossentropy_loss in [2.0](https://github.com/faneshion/MatchZoo/blob/2.0/matchzoo/losses.py#L36)

**Comment by logicmd** [comment]
What is dim_fan_out in 2.0? is it the size of q vector and d vector?

**Comment by bwanglzu** [comment]
1. we no longer have configurations in 2.0.
2. categorical_crossentropy is keras loss, while rank_crossentropy is matchzoo customized loss function for ranking task.
3. dim_fan_out means dimensional of the output layer.

Read [2.0 tutorial](https://nbviewer.jupyter.org/github/faneshion/MatchZoo/blob/2.0/notebooks/wikiqa/dssm.ipynb).

**Comment by logicmd** [comment]
Thanks @bwanglzu  I was following  2.0 tutorial.

**Comment by logicmd** [comment]
@bwanglzu I checked predictions
why it is a two column matrix?
```
In [13]: predictions2
Out[13]:
array([[0.9856516 , 0.01434833],
       [0.97185385, 0.02814611],
       [0.9855137 , 0.01448633],
       [0.9856515 , 0.0143485 ],
       [0.98537177, 0.0146282 ],
       [0.9855805 , 0.01441953],
       [0.9854152 , 0.01458476],
       [0.22783624, 0.77216375],
       [0.9856755 , 0.01432448],
       [0.98564583, 0.01435416],
       [0.9855576 , 0.01444245],
       [0.9854673 , 0.01453275],
       [0.98567605, 0.01432391],
       [0.21207833, 0.78792167],
       [0.9845259 , 0.01547402],
       [0.9856582 , 0.01434182],
       [0.98564607, 0.01435396],
       [0.9770775 , 0.02292252],
       [0.9856791 , 0.01432098],
       [0.5932053 , 0.4067948 ],
       [0.98543537, 0.01456465],
       [0.9856742 , 0.01432584],
       [0.98566157, 0.01433845],
       [0.21288468, 0.7871153 ],
       [0.98566276, 0.01433727],
       [0.97764194, 0.02235801],
       [0.9855759 , 0.01442412],
       [0.98563796, 0.01436202],
       [0.9786476 , 0.02135237],
       [0.9855019 , 0.01449806],
       [0.9856765 , 0.01432349],
       [0.9856755 , 0.01432446],
       [0.9856655 , 0.01433451],
       [0.98559046, 0.01440951],
       [0.98567563, 0.01432435],
       [0.98204076, 0.01795925],
       [0.9856593 , 0.01434074],
       [0.98567814, 0.01432182],
       [0.9855138 , 0.01448622],
       [0.9856645 , 0.01433554],
       [0.9856797 , 0.01432031],
       [0.98561513, 0.0143848 ],
       [0.98565173, 0.01434829],
       [0.9856515 , 0.01434848],
       [0.9856442 , 0.0143558 ],
       [0.98566633, 0.01433372],
       [0.21277227, 0.7872278 ],
       [0.9730958 , 0.02690428],
       [0.98567605, 0.01432396],
       [0.98562175, 0.01437822],
       [0.9856388 , 0.0143612 ],
       [0.9856712 , 0.01432884],
       [0.9856662 , 0.01433376],
       [0.9856712 , 0.01432883],
       [0.9856688 , 0.01433128],
       [0.98568   , 0.01432003],
       [0.9856629 , 0.0143371 ],
       [0.21267568, 0.7873243 ],
       [0.21215197, 0.78784806],
       [0.21237543, 0.78762454],
       [0.985664  , 0.01433595],
       [0.9831218 , 0.01687821],
       [0.98567975, 0.01432022],
       [0.98566425, 0.01433572]], dtype=float32)
```

**Comment by logicmd** [comment]
@bwanglzu maybe you guys update api but not tutorial, there is no more `ids` in `X_te` . I don't think it is worth a PR so I post it here.

```

for ids, pred in zip(X_te.ids, predictions):
    print("{} is predicted as {}".format(ids, pred))
```
should be 
```
for id_left, id_right, pred in zip(X_te.id_left, X_te.id_right, predictions):
    print("{} is predicted as {}".format((id_left, id_right), pred))

```

```

**Comment by bwanglzu** [comment]
@logicmd exactly, we sent a PR two days ago to update the generator. Would you like to send a PR to update the dssm tutorial?

**Comment by bwanglzu** [comment]
besides, how do you feel about the 2.0 api? We've been working on this for several months (part-time). Think all models will be there before the end of this year.

**Comment by logicmd** [comment]
sure, I can send a PR. The 2.0 API is much cleaner. To be honest there is too much parameter in 1.0, most of them I don't know how to correspond with CDSSM paper. For 2.0, I felt like parameter are much cleaner and it can be correspond with CDSSM paper easily.

**Comment by logicmd** [comment]
@bwanglzu could you help elaborate why prediction is two column matrix? which column should I trust?

**Comment by bwanglzu** [comment]
Hi, @logicmd as you might have seen my previous comment:

> The two column is a classification task, class 1 prob and class 2 prob. We're working on the metrics, accuracy is only for test at the moment.

if you want to create a ranking task, then you need to change the generator to ranking generator. for example:

```python
from matchzoo import tasks
from matchzoo import generator

// task type of pair generator is ranking by default.
pair_generator = generator.PairGenerator(inputs=train, num_neg=4, batch_size=64, stage='train', shuffle=True)
// or point generator
point_generator = generator.PointGenerator(inputs=train, task=tasks.Ranking(), batch_size=64, stage='train', shuffle=True)
```
Then you need to set the parameters as a ranking task.

```python
dssm_model = models.DSSMModel()
# do something
dssm_model.params['task'] = tasks.Ranking()
# do something
```
Then it will produce the 1d output

**Comment by bwanglzu** [comment]
@logicmd The bug has been fixed, see [here](https://github.com/faneshion/MatchZoo/blob/2.0/notebooks/wikiqa/dssm.ipynb) how to use dssm for ranking task.

**Comment by bwanglzu** [comment]
@logicmd Only `PairGenerator` has the parameter called `negative_sampling`.

**Comment by logicmd** [comment]
是的。请问何时用PairGenerator合适，何时用PointGenerator合适呀？CDSSM的分类任务中应该用PairGenerator吗？我因为在使用MatchZoo v1时候就生成了负例了，所以不希望framework再生成负例。

**Comment by bwanglzu** [comment]
@logicmd 当你需要正例匹配负例的时候用PairGenerator，如果已经弄好了直接用Point即可～

**Comment by uduse** [comment]
Closed due to inactivity.
